dev/connection_holder.h

```diff
@@ -12,11 +14,17 @@ namespace sqlite_orm {
 
         struct connection_holder {
 
-            connection_holder(std::string filename_) : filename(std::move(filename_)) {}
+            connection_holder(std::string filename_, vfs_mode vfs_, sqlite_orm::open_mode open_) :
+                filename(std::move(filename_)), vfs(vfs_), open(open_) {}
 
             void retain() {
                 if (1 == ++this->_retain_count) {
-                    auto rc = sqlite3_open(this->filename.c_str(), &this->db);
+
+                    const std::string vfs_name = internal::to_string(vfs);
```

Comment:
Should be `const serialize_result_type& vfs_name`, also see comment below for `to_string(vfs_mode v)`.